### PR TITLE
No attributes

### PR DESCRIPTION
**Pull request by sjpt**
Suggested resolution to https://github.com/mrdoob/three.js/issues/19430
I'm sure edits will be needed for code style, sorry
This can be tested from https://jsfiddle.net/sjpt/Lrhjkstz/44/
The fiddle refers to a different copy of my fork of three; I could not get CORS access to my github version.

### CODE REVIEW

**Comment by mrdoob** [comment]
It's hard to see what the changes here are:
https://github.com/mrdoob/three.js/pull/19440/files

**Comment by sjpt** [comment]
The changes are indeed very small, enough to ensure that `geometry.instanceCount` and `geometry.drawRange.count` are reflected in the arguments to `renderer.renderInstances( geometry, drawStart, drawCount, instanceCount )` when there are no instance or vertex attributes.  This permits use of pure procedural geometry (shader based with no vertex or instance attributes); without breaking previous attribute based geometry.

It also resolves a breaking change in revision 117; where unused attributes were ignored. In earlier versions where geometry had a 'dummy' position attribute unused in the shader the count of the attribute still established the count for rendering. After the change the unused position attribute was discarded **before** the count was established so the geometry no longer rendered.

Many more details in discussion at https://github.com/mrdoob/three.js/issues/19430

**Comment by xinaesthete** [comment]
I see what @mrdoob means about the relevant changes being difficult to see amidst the noise of whitespace changes etc.  I might try to make a cleaner version of this PR as I believe rendering pure procedural geometry without vertex attributes is a valuable addition to the library.

**Comment by Mugen87** [comment]
@sjpt Can you please remove all build files from your PR first? 

Besides, I think we need a new WebGL 2 example that demonstrates this new feature and makes it easier to understand for interested developers. Maybe something similar like from this article:

https://webgl2fundamentals.org/webgl/lessons/webgl-drawing-without-data.html

File: src/renderers/WebGLRenderer.js [hunk 1]
```diff
@@ -829,43 +835,43 @@ function WebGLRenderer( parameters ) {
...
 
 		} else if ( object.isPoints ) {
 
-			renderer.setMode( _gl.POINTS );
+			renderer.setMode( 0 );
 
 		} else if ( object.isSprite ) {
 
-			renderer.setMode( _gl.TRIANGLES );
+			renderer.setMode( 4 );
```
Comment (Mugen87) [inline src/renderers/WebGLRenderer.js:874]: It seems not necessary to change the calls of `setMode()`.

File: src/renderers/WebGLRenderer.js [hunk 2]
```diff
@@ -799,15 +805,15 @@ function WebGLRenderer( parameters ) {
 
 			if ( index !== null ) {
 
-				_gl.bindBuffer( _gl.ELEMENT_ARRAY_BUFFER, attribute.buffer );
+				_gl.bindBuffer( 34963, attribute.buffer );
```
Comment (Mugen87) [inline src/renderers/WebGLRenderer.js:808]: Better to use the `_gl.ELEMENT_ARRAY_BUFFER` here.

**Comment by sjpt** [comment]
> I think we need a new very simple WebGL 2 example that demonstrates this new feature and makes it easier to understand for interested developers. Maybe something similar like from this article:
> 
> https://webgl2fundamentals.org/webgl/lessons/webgl-drawing-without-data.html

I've made a version that uses his circle example. I have not included his aspect ratio correction as it does not directly impact this issue, but have retained the use of multiple instances (without instance buffer)

**Comment by Mugen87** [comment]
You now have removed the build files and `package.json` related files. The idea was to just undo the *changes*.

**Comment by sjpt** [comment]
Sorry, trying to get an understanding of git. I was trying to remove the files from the pull request.

**Comment by sjpt** [comment]
I've created a new pull request
https://github.com/mrdoob/three.js/pull/19451

The essential changes are the same. Some change marker comments removed, 
Changes to replace some constants to symbolic such as _gl.ELEMENT_ARRAY_BUFFER merged.

**Comment by Mugen87** [comment]
Closing in favor of #19451.

**Pull request closed without merging**